### ISSUE TITLE
How to avoid malicious node fork to skip previous miner's block

### ISSUE THREAD

**Issue by fxsjy** [virtualeconomy/v-systems#88]
If a miner A is before miner B, C, and A is hated by B and C.

After A has produced a new block called "ABlock", B and C what the "ABlock" be ignored. 

Then , B produces a block which is attached to the previous block of "ABlock", then C will produce the next block after B's block

    Block N <---- ("ABlock": Block of A)
                     |
                     +--- (Block of B) <----- (Block of C)

As a result, A's block will be ignored or rollbacked.

How does spos handle this case?

**Comment by ncying** [comment]
SPoS is consensus focus on **community autonomy**, which means it will give the right to all users.

As the case you mention above, if minter A's minting average balance (MAB) is less than the sum of minter B and C's MAB, the blocks generated by A will be rollbacked.

For A, the best choice is to improve it's MAB (attract more community users to lease their balance) to avoid this situation. And it can also report the related issue to the community. Base on the report the community will reduce the MAB (by canceling lease the VSYS from the minter) or prepare another minter to replace B and C.

When B and C contended by other minters, their addresses can not be minter again. And they need to raise another huge number of VSYS from the community. Furthermore, they also need extra cost in contend fee (50000VSYS) and fee to maintain the node and backup.

In summary, in SPoS systems, it encourages all minters not to be Evil. Since the RESULT of "bad actions" is shared by the whole community. (SPoS systems all transaction fee is destroyed, which means reward to every user in the community)

**Comment by fxsjy** [comment]
Thanks you, very clear answer.
One more question, who will receive the contend fee, or the fee is just be burnt?

**Comment by ncying** [comment]
As mentioned in the last comment, **ALL** tx fee will be destroyed directly, include the `contend fee`. In another view, it means the tx fee is shared by all community users(by reducing the inflation of the systems).

**Comment by fxsjy** [comment]
@ncying, for the mecanism:  "if minter A's minting average balance (MAB) is less than the sum of minter B and C's MAB, the blocks generated by A will be rollbacked."

Could you please tell me in which source file it is implemented?   I want to study it deeply.

Thank you very much.

**Comment by ncying** [comment]
1. In each block, there is a feature called `score = max(MAB/100000000, 1L)` (1e8 is the unit of VSYS)
2. when append a block, a cumulated score will be calculated
3. when process the fork, this score will be used

It is not in one particular source file, it was implement in several functions which distributed in several files.

**Comment by fxsjy** [comment]
@ncying , thank you. I understand it more clearly now.
